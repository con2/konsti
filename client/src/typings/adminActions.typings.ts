import { Game } from 'typings/game.typings';

export const SUBMIT_UPDATE_HIDDEN = 'SUBMIT_UPDATE_HIDDEN';
export const SUBMIT_GET_SETTINGS = 'SUBMIT_GET_SETTINGS';
export const SUBMIT_SELECT_SIGNUP_TIME = 'SUBMIT_SELECT_SIGNUP_TIME';
export const SUBMIT_SET_TEST_TIME = 'SUBMIT_SET_TEST_TIME';
export const SUBMIT_TOGGLE_APP_OPEN = 'SUBMIT_TOGGLE_APP_OPEN';
export const SUBMIT_RESPONSE_MESSAGE = 'SUBMIT_RESPONSE_MESSAGE';

export interface SubmitUpdateHiddenAsync {
  type: typeof SUBMIT_UPDATE_HIDDEN;
  hiddenGames: readonly Game[];
}

export interface SubmitGetSettingsAsync {
  type: typeof SUBMIT_GET_SETTINGS;
  hiddenGames: readonly Game[];
  signupTime: string;
  appOpen: boolean;
}

export interface SubmitSignupTimeAsync {
  type: typeof SUBMIT_SELECT_SIGNUP_TIME;
  signupTime: string;
}

export interface SubmitSetTestTime {
  type: typeof SUBMIT_SET_TEST_TIME;
  testTime: string;
}

export interface SubmitToggleAppOpenAsync {
  type: typeof SUBMIT_TOGGLE_APP_OPEN;
  appOpen: boolean;
}

export interface SubmitResponseMessageAsync {
  type: typeof SUBMIT_RESPONSE_MESSAGE;
  responseMessage: string;
}

export type AdminActionTypes =
  | SubmitUpdateHiddenAsync
  | SubmitGetSettingsAsync
  | SubmitSignupTimeAsync
  | SubmitSetTestTime
  | SubmitToggleAppOpenAsync
  | SubmitResponseMessageAsync;
