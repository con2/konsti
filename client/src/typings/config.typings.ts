export interface Config {
  appName: string;
  SIGNUP_END_TIME: number;
  SIGNUP_OPEN_TIME: number;
  MESSAGE_DELAY: number;
  CONVENTION_NAME: ConventionName;
  CONVENTION_YEAR: string;
  CONVENTION_START_TIME: string;
  DAY_START_TIME: number;
  apiServerURL: string;
  useTestTime: boolean;
  reduxTrace: boolean;
  loadedSettings: string;
  enableAxe: boolean;
  enableWhyDidYouRender: boolean;
  dataUpdateInterval: number;
  noSignupGames: string[];
  revolvingDoorEnabled: boolean;
  tagFilteringEnabled: boolean;
  simpleDetails: boolean;
}

type ConventionName = 'Ropecon' | 'Tracon Hitpoint';
