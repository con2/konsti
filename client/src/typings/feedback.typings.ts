export interface Feedback {
  feedback: string;
  gameId: string;
}

export interface PostFeedbackResponse {
  message: string;
  status: 'success';
}
