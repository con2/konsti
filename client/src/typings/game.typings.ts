export interface Game {
  gameId: string;
  title: string;
  description: string;
  location: string;
  startTime: string;
  mins: number;
  tags: readonly string[];
  genres: readonly string[];
  styles: readonly string[];
  language: string;
  endTime: string;
  people: string;
  minAttendance: number;
  maxAttendance: number;
  gameSystem: string;
  englishOk: boolean;
  childrenFriendly: boolean;
  ageRestricted: boolean;
  beginnerFriendly: boolean;
  intendedForExperiencedParticipants: boolean;
  shortDescription: string;
  revolvingDoor: boolean;
  popularity: number;
  programType: string;
}

export interface DnDUpdatedPositions {
  availableGames?: readonly Game[];
  selectedGames?: readonly Game[];
}

export interface DnDMove {
  index: number;
  droppableId: string;
}

export interface PostGamesUpdateResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface GetGamesResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface PostHiddenResponse {
  hiddenGames: Game[];
  message: string;
  status: 'success';
}
