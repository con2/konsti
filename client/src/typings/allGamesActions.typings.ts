import { Game } from 'typings/game.typings';

export const SUBMIT_GET_GAMES = 'SUBMIT_GET_GAMES';

export interface SubmitGetGamesAsync {
  type: typeof SUBMIT_GET_GAMES;
  games: readonly Game[];
}

export type AllGamesActionTypes = SubmitGetGamesAsync;
