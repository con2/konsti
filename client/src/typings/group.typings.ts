import { Signup } from 'typings/user.typings';

export interface GroupMember {
  enteredGames: readonly Signup[];
  groupCode: string;
  serial: string;
  signedGames: readonly Signup[];
  username: string;
}

export interface GroupData {
  groupCode: string;
  leader: boolean;
  ownSerial: string;
  username: string;
  leaveGroup?: boolean;
  closeGroup?: boolean;
}

export interface PostGroupResponse {
  groupCode: string;
  message: string;
  status: 'success';
}

export interface GetGroupResponse {
  message: string;
  results: GroupMember[];
  status: 'success';
}
