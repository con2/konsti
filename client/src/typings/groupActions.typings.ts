import { GroupMember } from 'typings/group.typings';

export const SUBMIT_UPDATE_GROUP_CODE = 'SUBMIT_UPDATE_GROUP_CODE';
export const SUBMIT_LEAVE_GROUP = 'SUBMIT_LEAVE_GROUP';
export const SUBMIT_UPDATE_GROUP_MEMBERS = 'SUBMIT_UPDATE_GROUP_MEMBERS';

export interface SubmitUpdateGroupCodeAsync {
  type: typeof SUBMIT_UPDATE_GROUP_CODE;
  groupCode: string;
}

export interface SubmitGetGroupAsync {
  type: typeof SUBMIT_UPDATE_GROUP_MEMBERS;
  groupMembers: readonly GroupMember[];
}

export interface SubmitLeaveGroupAsync {
  type: typeof SUBMIT_LEAVE_GROUP;
  groupCode: string;
}

export type GroupActionTypes =
  | SubmitUpdateGroupCodeAsync
  | SubmitGetGroupAsync
  | SubmitLeaveGroupAsync;
