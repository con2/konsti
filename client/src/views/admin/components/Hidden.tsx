import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import _ from 'lodash';
import { timeFormatter } from 'utils/timeFormatter';
import { Game } from 'typings/game.typings';

export interface Props {
  hiddenGames: readonly Game[];
}

export const Hidden: FC<Props> = (props: Props): ReactElement => {
  const { hiddenGames } = props;
  const { t } = useTranslation();

  const sortedGames = _.sortBy(hiddenGames, [
    (hiddenGame) => hiddenGame.title.toLowerCase(),
  ]);

  const GamesList = sortedGames.map((game) => (
    <li key={game.gameId}>
      <Link to={`/games/${game.gameId}`}>{game.title}</Link>
      {' - '}
      {timeFormatter.weekdayAndTime({
        time: game.startTime,
        capitalize: false,
      })}
    </li>
  ));

  return (
    <div className='hidden'>
      <h3>{t('hiddenGames')}</h3>
      <ul>
        {!hiddenGames && <span>{t('noHiddenGames')}</span>}
        {GamesList}
      </ul>
    </div>
  );
};
