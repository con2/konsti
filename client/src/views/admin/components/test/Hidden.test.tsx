import React from 'react';
import { shallow } from 'enzyme';
import { Hidden, Props } from '../Hidden';
import { Game } from 'typings/game.typings';

const hiddenGames: Game[] = [];

describe('Hidden', () => {
  it('should render correctly', () => {
    const props: Props = { hiddenGames };
    const component = shallow(<Hidden {...props} />);
    expect(component).toMatchSnapshot();
  });
});
