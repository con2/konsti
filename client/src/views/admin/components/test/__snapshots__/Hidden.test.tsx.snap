// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hidden should render correctly 1`] = `
<div
  className="hidden"
>
  <h3>
    hiddenGames
  </h3>
  <ul />
</div>
`;
